hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -734,6 +806,11 @@ public boolean mkdirs(String src, FsPermission masked, boolean createParent)
         new Class<?>[] {String.class, FsPermission.class, boolean.class},
         new RemoteParam(), masked, createParent);
 
+    // Auto-create user home dir for a trash path.
+    // moveToTrash() will first call fs.mkdirs() to create the parent dir, before calling rename()
+    // to move the file into it. As a result, we need to create user home dir in mkdirs().
+    createUserHomeForTrashPath(src, locations);
```

Comment:
Should this be a flag so we enable it via config?

Reply:
I am open to that option. If we control with a flag, should this be turned on by default or should it be off?

Reply:
added a config to control it.